packages/@lwc/template-compiler/src/parser/attribute.ts

```diff
@@ -121,10 +121,15 @@ export function normalizeAttributeValue(
         }
     }
 
-    const isQuoted = isQuotedAttribute(raw);
-    const isEscaped = isEscapedAttribute(raw);
-    if (!isEscaped && isExpression(value)) {
-        if (isQuoted) {
+    const isQuoted = isQuotedAttribute(rawValue);
+    const matchedBrackets = hasMatchingBrackets(rawValue);
+    if (matchedBrackets && isExpression(value)) {
```

Comment:
Do we need to explicitly check for matched brackets when matched brackets are a prerequisite for expressions?